compio-dispatcher/src/lib.rs

```diff
@@ -6,28 +6,37 @@ use std::{
     future::Future,
     io,
     num::NonZeroUsize,
-    panic::resume_unwind,
+    panic::{resume_unwind, UnwindSafe},
+    sync::{Arc, Mutex},
     thread::{available_parallelism, JoinHandle},
 };
 
-use crossbeam_channel::{unbounded, SendError, Sender};
+use compio_driver::{AsyncifyPool, ProactorBuilder};
+use compio_runtime::event::{Event, EventHandle};
+use crossbeam_channel::{unbounded, Sender};
 use futures_util::{future::LocalBoxFuture, FutureExt};
 
-type BoxClosure<'a> = Box<dyn (FnOnce() -> LocalBoxFuture<'a, io::Result<()>>) + Send>;
-
 /// The dispatcher. It manages the threads and dispatches the tasks.
 pub struct Dispatcher {
-    sender: Sender<BoxClosure<'static>>,
-    threads: Vec<JoinHandle<io::Result<()>>>,
+    sender: Sender<DispatcherClosure>,
+    threads: Vec<JoinHandle<()>>,
+    pool: Arc<AsyncifyPool>,
```

Comment:
Why does `AsyncifyPool` need to be shared with Arc? Seems its fields are all cheap clonable